src/java.base/share/native/libjava/ClassLoader.c

```diff
@@ -99,7 +99,14 @@ Java_java_lang_ClassLoader_defineClass1(JNIEnv *env,
         return 0;
     }
 
+    // On malloc(0), implementators of malloc(3) have the choice to return either
+    // NULL or a unique non-NULL pointer. To unify libc behavior across our platforms
+    // we chose the latter. (see 8283225)
+    #ifdef _AIX
+    body = (jbyte *)malloc(length == 0 ? 1 : length);
```

Comment:
Using AIX_ONLY this can be simplified:

`body = (jbyte *)malloc(length AIX_ONLY( == 0 ? 1 : length));`

Reply:
> Using AIX_ONLY this can be simplified:
> 
> `body = (jbyte *)malloc(length AIX_ONLY( == 0 ? 1 : length));`

This is jdk, not hotspot. Do we have AIX_ONLY in the JDK?

Reply:
As Thomas mentioned, this does not seem to work here.